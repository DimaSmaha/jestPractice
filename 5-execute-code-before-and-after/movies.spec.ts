import Movies from "./movies";

describe("Movies", () => {
  let movies: Movies;
  beforeAll(() => {
    movies = new Movies();
    movies.addMovie("Jumanji");
  });

  test("Test Add Movie", () => {
    movies.addMovie("Fast & Furious");

    expect(movies.getCollection()).toMatchSnapshot();
    // expect(movies.getCollection()).toEqual([
    //   { title: "Jumanji", rating: null },
    //   { title: "Fast & Furious", rating: null },
    // ]);
  });

  test("Test Add Rating", () => {
    movies.rateMovie(0, 5);

    expect(movies.getCollection()).toMatchSnapshot();
    // expect(movies.getCollection()).toEqual([
    //   { title: "Jumanji", rating: 5 },
    //   { title: "Fast & Furious", rating: null },
    // ]);
  });
});
