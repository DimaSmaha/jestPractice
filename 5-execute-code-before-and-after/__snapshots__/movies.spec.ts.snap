// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies Test Add Movie 1`] = `
[
  {
    "rating": null,
    "title": "Jumanji",
  },
  {
    "rating": null,
    "title": "Fast & Furious",
  },
]
`;

exports[`Movies Test Add Rating 1`] = `
[
  {
    "rating": 5,
    "title": "Jumanji",
  },
  {
    "rating": null,
    "title": "Fast & Furious",
  },
]
`;
